spec/requests/api/suppliers_controller_locations_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::SuppliersController do
+  let(:access_token) { 'spoofed-token' }
+  let(:headers) { { 'CONTENT_TYPE': content_type, 'Authorization': "Bearer #{access_token}" } }
+  let(:content_type) { ApiController::CONTENT_TYPE }
+  let(:response_json) { JSON.parse(response.body) }
+
+  describe 'GET /suppliers/:supplier_id/locations' do
+    let(:supplier) { create(:supplier) }
+
+    it 'returns the locations' do
+      location1 = create(:location, suppliers: [supplier])
+      location2 = create(:location, suppliers: [supplier])
```

Comment:
Maybe add a another location that's not linked to a move - or maybe linked only via `to_location` to demonstrate that the move `from_location` is the only thing that associates a move to a supplier. Perhaps add a dedicated spec for this just to make the behaviour crystal clear.